File: CLIENT/src/types/StockMovement.ts
import { User } from "./User";

export type StockMovement = {
    id: string,
    productId: string,
    quantityBefore: number,
    quantityAfter: number,
    priceBefore: number,
    priceAfter: number,
    createdBy: string | null,
    createdAt: Date,
    status: StockMovementStatus,
    decidedAt: Date | null,
    user: User
};

enum StockMovementStatus
{
    PENDING = 1,
    ACCEPTED = 2,
    DENIED = 3
}

File: API/Entities/StockMovement.cs
using System;

namespace API.Entities;

public class StockMovement
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public required string ProductId { get; set; }
    public int QuantityBefore { get; set; }
    public int QuantityAfter { get; set; }
    public int PriceBefore { get; set; }
    public int PriceAfter { get; set; }
    public string? CreatedBy { get; set; } 
    public AppUser User { get; set; } = default!;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public StockMovementStatus Status { get; set; } = StockMovementStatus.PENDING;
    public DateTime? DecidedAt { get; set; }
}

public enum StockMovementStatus
{
    PENDING = 1,
    ACCEPTED = 2,
    DENIED = 3
}